RecoLocalCalo/HcalRecProducers/plugins/alpaka/Mahi.dev.cc

```diff
@@ -0,0 +1,1460 @@
+          41227.5, 43275.5, 45323.5, 47371.5, 49419.5, 51467.5, 53515.5, 55563.5, 57611.5, 59659.5, 61707.5, 63755.5,
+          65803.5, 67851.5, 69899.5, 71947.5, 73995.5, 76043.5, 78091.5, 80139.5, 82187.5, 84235.5, 88331.5, 92427.5,
+          96523.5, 100620,  104716,  108812,  112908};
+
+      constexpr int32_t IPHI_MAX = 72;
+
+      ALPAKA_FN_ACC ALPAKA_FN_INLINE uint32_t did2linearIndexHB(
+          uint32_t const didraw, int const maxDepthHB, int const firstHBRing, int const lastHBRing, int const nEtaHB) {
+        HcalDetId did{didraw};
+        uint32_t const value = (did.depth() - 1) + maxDepthHB * (did.iphi() - 1);
+        return did.ieta() > 0
+                   ? value + maxDepthHB * hcal::reconstruction::mahi::IPHI_MAX * (did.ieta() - firstHBRing)
+                   : value + maxDepthHB * hcal::reconstruction::mahi::IPHI_MAX * (did.ieta() + lastHBRing + nEtaHB);
+      }
+      ALPAKA_FN_ACC ALPAKA_FN_INLINE uint32_t did2linearIndexHE(uint32_t const didraw,
+                                                                int const maxDepthHE,
+                                                                int const maxPhiHE,
+                                                                int const firstHERing,
+                                                                int const lastHERing,
+                                                                int const nEtaHE) {
+        HcalDetId did{didraw};
+        uint32_t const value = (did.depth() - 1) + maxDepthHE * (did.iphi() - 1);
+        return did.ieta() > 0 ? value + maxDepthHE * maxPhiHE * (did.ieta() - firstHERing)
+                              : value + maxDepthHE * maxPhiHE * (did.ieta() + lastHERing + nEtaHE);
+      }
+      ALPAKA_FN_ACC ALPAKA_FN_INLINE uint32_t get_qiecoder_index(uint32_t const capid, uint32_t const range) {
+        return capid * 4 + range;
+      }
+
+      ALPAKA_FN_ACC ALPAKA_FN_INLINE float compute_reco_correction_factor(float const par1,
+                                                                          float const par2,
+                                                                          float const par3,
+                                                                          float const x) {
+        return par3 * x * x + par2 * x + par1;
+      }
+
+      // compute the charge using the adc, qie type and the appropriate qie shape array
+      ALPAKA_FN_ACC ALPAKA_FN_INLINE float compute_coder_charge(
+          int const qieType, uint8_t const adc, uint8_t const capid, float const* qieOffsets, float const* qieSlopes) {
+        auto const range = qieType == 0 ? (adc >> 5) & 0x3 : (adc >> 6) & 0x3;
+        auto const* qieShapeToUse = qieType == 0 ? qie8shape : qie11shape;
+        auto const nbins = qieType == 0 ? 32 : 64;
+        auto const center = adc % nbins == nbins - 1 ? 0.5 * (3 * qieShapeToUse[adc] - qieShapeToUse[adc - 1])
```

Comment:
### TODO for later
This is going to be in `double`, because of `0.5`, but I don't think we need that much precision.
Would it be OK to use `0.5f` and store the result explicitly as a `float` ?